keps/sig-node/24-apparmor/README.md

```diff
@@ -0,0 +1,593 @@
+provide such enhanced functionality on top.
+
+###### Profiles managed by the cluster admins
+
+The current support relies on profiles being loaded on all cluster nodes
+where the pods using them may be scheduled. It is also the cluster admin's
+responsibility to ensure the profiles are correctly saved and synchronized
+across the all nodes. Existing mechanisms like node `labels` and `nodeSelectors`
+can be used to ensure that pods are scheduled on nodes supporting their desired
+profiles.
+
+##### RuntimeDefault Profile
+
+We propose maintaining the support to a single runtime profile, which will be
+defined by using the `AppArmorProfileTypeRuntimeDefault`. The reasons being:
+
+- No changes to the current behavior. Users are currently not allowed to specify
+  other runtime profiles. The existing API server rejects runtime profile names
+  that are different than `runtime/default`.
+- Most runtimes only support the default profile, although the CRI is flexible
+  enough to allow the kubelet to send other profile names.
+- Multiple runtime profiles has never been requested as a feature.
+
+If built-in support for multiple runtime profiles is needed in the future, a new
+KEP will be created to cover its details.
+
+#### Validation
+
+Current annotation validations:
+- Pod annotations are immutable (cannot be added, modified, or removed on pod update)
+- Annotation value must have a `localhost/` prefix, or be one of: `""`, `runtime/default`, `unconfined`.
+
+The following additional validations are proposed:
+1. Fields must match the corresponding annotations when both are present, except for ephemeral containers.
+2. AppArmor profile must be unset on Windows pods (`spec.os.name == "windows"`). Only enforced on fields.
+3. Localhost profile must not be empty, and must not be padded with whitespace. Only enforced on creation.
+   This was previously enforced by the [Kubelet](https://github.com/kubernetes/kubernetes/blob/2624e93d55375a9642977d4d5795841ab7463b1d/pkg/security/apparmor/validate.go#L70-L77).
+
+*Note on localhost profile validation:* AppArmor profile naming is flexible, but both of the leading
+CRI implementations (containerd & cri-o) require a profile with a matching name to be loaded. This
+prevents the special `unconfined` profile, or various wildcard and variable profile names from being
+used in practice. This validation is deferred to the runtime, rather than being enforced by the API
+for backwards compatibility.
+
+#### Node Status
+
+The Kubelet currently appends the AppArmor status to the node ready condition message:
+
+https://github.com/kubernetes/kubernetes/blob/5c664599a9a80f01c30d924702f7ad799f7dd956/pkg/kubelet/nodestatus/setters.go#L530-L536
+
+The ready condition is certainly not the right place for this message, but more generally the
+Kubelet does not broadcast the status of every optional feature. For GA, this message will simply be
```

Comment:
need to make sure to update relonotes in case there are any dependencines on this.